### PR TITLE
Add details about custom 404

### PR DESCRIPTION
**Pull request by tinymachine**
- include option of creating a `404.md` page in addition to `404.astro`
- mention that custom 404 won't work by default in dev environment

#### What kind of changes does this PR include?

- [ ] Minor content fixes (broken links, typos, etc.)
- [X] New or updated content
- [ ] Translated content
- [ ] Changes to the docs site code
- [ ] Something else!

#### Description

- Added additional details to clarify how the custom 404 works.

### CODE REVIEW

**Comment by delucis** [comment]
Thanks for this! Can we double check that the dev server behaviour is still broken? https://github.com/withastro/astro/pull/3331 released in v1.0.0-beta.28 is supposed to have fixed that.

**Comment by sarah11918** [comment]
And heads up, v1.0.0-beta.28 has been rolled back to .27, so even if behaviour is fixed there, it might still take a little bit of time to confirm. 

So, I'm going to label this as something we need to confirm before adding to the docs, but don't worry, we know it's here!

**Comment by tinymachine** [comment]
Glad to hear it's a bug! Thanks for keeping an eye on this.

**Comment by delucis** [comment]
Great — I’d missed that rollback to beta.27. So we can hold fire and 🤞 hopefully all this PR will need to do is mention you can use a `404.md` once the next beta is released.

**Comment by tony-sull** [comment]
`beta.29` was released this morning with the dev server fix 🎉 

Markdown pages aren't actually supported for custom 404 pages right now though 👀 I just filed [an issue](https://github.com/withastro/astro/issues/3389) to fix that and will actually be working on routing bugs today so this should be unblocked soon!

**Comment by sarah11918** [comment]
I see that the bug and fix seem to be merged @tony-sull , so is it safe to assume that the new wording for docs is fine, minus the warning about it not working in dev? Are we waiting for anything else?

File: src/pages/en/core-concepts/astro-pages.md
Comment (delucis) [inline src/pages/en/core-concepts/astro-pages.md:129]: Now that https://github.com/withastro/astro/pull/3331 has been released we can remove this warning.

```suggestion
```

**Review by delucis — CHANGES_REQUESTED** [review]
As @sarah11918 noted, since https://github.com/withastro/astro/pull/3390 has been released, Markdown pages work so we can merge this once the warning is removed.

File: src/pages/en/core-concepts/astro-pages.md
Comment (tinymachine) [reply src/pages/en/core-concepts/astro-pages.md:129]: Oh sorry — should I have taken action on this? I'll hit the 'Commit suggestion' button above and see what happens! 😅🤞

Comment (sarah11918) [reply src/pages/en/core-concepts/astro-pages.md]: 😅 

This is great, thanks! 

As Maintainers, Chris and I *can* commit it ourselves, but we like to give people the chance to do it themselves. And, the more comfortable you feel doing stuff, the easier it is when you contribute **next time**!  ;)

File: src/pages/en/core-concepts/astro-pages.md
Comment (delucis) [inline src/pages/en/core-concepts/astro-pages.md:127]: I’m going to take an executive decision to maintain the whitespace as it was prior to the PR now that we don’t have the separate blockquote here and get this merged. Thanks @tinymachine 🙌 

```suggestion
For a custom 404 error page, you can create a `404.astro` or `404.md` file in `/src/pages`.

  This will build to a `404.html` page. Most [deploy services](/en/guides/deploy/) will find and use it.
```

File: src/pages/en/core-concepts/astro-pages.md
Comment (tinymachine) [reply src/pages/en/core-concepts/astro-pages.md]: Ah, thanks for the teachable moment! 😄  Did I do that right, or is there something left for me to do? (I'm not that experienced with PRs!)

**Review by delucis — APPROVED** [review]

**Pull request merged**
